### ISSUE TITLE
When using ajax: function(data,callback,settings){} there is no way to specify the ajax.dataSrc option.

### ISSUE THREAD

**Issue by kidplug** [DataTables/DataTables#867]
If I have implemented my own ajax function to retrieve data, the ability to specify the "dataSrc" should still be available somehow.

Your code only reads the ajax.dataSrc if ajax is a plan object.
```
var dataSrc = $.isPlainObject( oSettings.ajax ) && oSettings.ajax.dataSrc !== undefined ?
			oSettings.ajax.dataSrc :
			oSettings.sAjaxDataProp; // Compatibility with 1.9-.
```

Then it falls back on sAjaxDataProp which seems likely to be deprecated in the future.

**Comment by DataTables** [comment]
> If I have implemented my own ajax function to retrieve data, the ability to specify the "dataSrc" should still be available somehow.

I'm afraid I don't agree - sorry! If you are defining a function then you need to pass the data array into the `callback` function - that could be as simple as `callback( json.data );`, or the array might be dynamically created from an object, or whatever else.

Providing a way to use `ajax.dataSrc` with a function seems redundant to me - in the case above you would just be setting it to be `data` anyway, so is it not easier just do that in your own function?

**Comment by kidplug** [comment]
I agree with your 2nd statement above but that's not how your code is working I think.

Your example `callback (json.data);` doesn't work as expected.
The `callback` function is expecting an object equivalent to the full ajax response- with the array (aka rows) included as a "data" property.

This doesn't work:
```
callback( myArray );
```

This does work:
```
var obj = { data: myArray };
callback(obj);
```

It'd be better IMO if as you say, the implemented ajax function could simply provide the actual array of data (ie rows) rather than having to return an outer object with the "data" property set to the array of rows.

My point with this issue was that I wanted to be able to do something like this:
```
var obj = {  users:  myArray  }
callback( obj, {  dataSrc: "users"   } );
```

But I'd really rather just be able to do `callback(myArray);`

Thanks.

**Comment by DataTables** [comment]
Sorry - you are correct - it does expect the object. The reason it has to do that is that it needs to be able to get the server-side processing parameters if that is enabled.

I agree - this should accept an array as well as the object and use some internal logic to switch as needed between whatever the callback is given. Reopening.
